lgc/patch/Gfx6ConfigBuilder.cpp

```diff
@@ -53,12 +53,10 @@ void ConfigBuilder::buildPalMetadata() {
     buildPipelineCsRegConfig();
   else {
     const bool hasTs = (m_hasTcs || m_hasTes);
+    const bool fsOnly = !m_pipelineState->isWholePipeline() && m_pipelineState->hasShaderStage(ShaderStageFragment);
 
-    if (!m_pipelineState->isWholePipeline() && m_pipelineState->hasShaderStage(ShaderStageFragment)) {
-      // FS-only shader compilation (part-pipeline compilation)
-      buildPipelineVsFsRegConfig();
-    } else if (!hasTs && !m_hasGs) {
-      // VS-FS pipeline
+    if ((!hasTs && !m_hasGs) || fsOnly) {
```

Comment:
Isn't the `fsOnly` check redundant even? If `fsOnly` is true, then we should always have `!hasTs && !m_hasGs` -- right? (And of course the same would be true in gfx9+)

Reply:
Yes, I find it is redundant. The code path is naturally to be a special VS-FS pipeline. Will remove it.